e2e/test/scenarios/admin-2/api-keys.cy.spec.ts

```diff
@@ -0,0 +1,371 @@
+    });
+
+    cy.findByRole("listbox").findByText("Administrators").click();
+
+    cy.findByLabelText("Create a new API Key").button("Create").click();
+
+    cy.wait("@createKey");
+    cy.wait("@getKeys");
+
+    cy.findByLabelText("Copy and save the API key").findByLabelText(
+      /the api key/i,
+    );
+
+    cy.button("Done").click();
+    cy.findByTestId("api-keys-table").findByText("New key");
+  });
+
+  it("should allow deleting an API key", () => {
+    createApiKey("Test API Key One", ALL_USERS_GROUP_ID);
+    cy.visit("/admin/settings/authentication/api-keys");
+    cy.wait("@getKeys");
+
+    cy.findByTestId("api-keys-table")
+      .contains("Test API Key One")
+      .closest("tr")
+      .icon("trash")
+      .click();
+    cy.findByLabelText("Delete API Key").button("Delete API Key").click();
+
+    cy.wait("@deleteKey");
+    cy.wait("@getKeys");
+
+    cy.findByTestId("api-keys-table").should("not.contain", "Test API Key One");
+    cy.findByTestId("api-keys-table").findByText("No API keys here yet");
+  });
+
+  it("should allow editing an API key", () => {
+    createApiKey("Development API Key", ALL_USERS_GROUP_ID);
+    cy.visit("/admin/settings/authentication/api-keys");
+    cy.wait("@getKeys");
+
+    cy.findByTestId("api-keys-table")
+      .should("include.text", "Development API Key")
+      .and("include.text", "All Users");
+
+    cy.findByTestId("api-keys-table").icon("pencil").click();
+
+    cy.findByLabelText(/Key name/)
+      .clear()
+      .type("Different key name");
+
+    cy.findByLabelText(/group/).click();
+    cy.findByRole("listbox").findByText("collection").click();
+
+    cy.button("Save").click();
+    cy.wait("@updateKey");
+    cy.wait("@getKeys");
+
+    cy.findByTestId("api-keys-table")
+      .should("not.contain", "Development API Key")
+      .contains("Different key name")
+      .closest("tr")
+      .should("contain", "collection");
+  });
+
+  it("should allow regenerating an API key", () => {
+    createApiKey("Personal API Key", ALL_USERS_GROUP_ID);
+
+    cy.visit("/admin/settings/authentication/api-keys");
+    cy.wait("@getKeys").then(({ response }) => {
+      const { created_at, updated_at } = response?.body[0];
+      // on creation, created_at and updated_at should be the same
+      expect(created_at).to.equal(updated_at);
+    });
+    cy.findByTestId("api-keys-table")
+      .contains("Personal API Key")
+      .closest("tr")
+      .icon("pencil")
+      .click();
+    cy.button("Regenerate API Key").click();
+    cy.button("Regenerate").click();
+    cy.wait("@regenerateKey");
+    cy.findByLabelText("The API key").should("include.value", "mb_");
+
+    cy.wait("@getKeys").then(({ response }) => {
+      const { created_at, updated_at } = response?.body[0];
+      // after regeneration, created_at and updated_at should be different
+      // they're too close to check via UI though
+      expect(created_at).to.not.equal(updated_at);
```

Comment:
I'm just thinking out loud, and this might be well known but I'll mention it anyway: Cypress's `equal` method checks for strict equality, so the following test passes:

`expect(new Date("2024-01-05")).to.not.equal(new Date("2024-01-05"));`

The silliness of this is pretty vivid if you look at Cypress's output:
<img width="352" alt="image" src="https://github.com/metabase/metabase/assets/130925/d24e90e6-031b-4c56-bcf1-8fa5fc36c7fb">

Since `created_at` and `updated_at` are strings, the `not.equal` pattern works perfectly well on line 180 above, but I think the pattern should perhaps be viewed with a bit of skepticism.

This is an alternative, but it's probably overkill:

`expect(new Date(created_at).getTime()).to.not.equal(new Date(updated_at).getTime());`

Even this has an amusing drawback: if both variables are `undefined`, the assertion will pass! `new Date(undefined).getTime()` is `NaN` and, bless JS, `NaN !== NaN`. To avoid this one could add `expect(new Date(created_at).getTime()).should('be.gt', 0)`

Reply:
Good to know, but I still feel pretty good about this check since we're not dealing with objects at all in api responses.